runbot_travis2docker/models/runbot_branch.py

```diff
@@ -4,13 +4,33 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 import re
+import os
 
 import subprocess
 import requests
 
 from openerp import fields, models, api, tools
 
 
+def _ssh_keyscan(ssh):
+    cmd = "ssh-keyscan -p %s %s"
+    match = re.search(r'@(?P<port_host>[^/]+)', ssh)
+    if match:
+        port_host = match.groupdict()['port_host'].split(':')
+        host, port = ((port_host[0], 22) if len(port_host) == 1 else
+                      (port_host[0], port_host[1]))
+        cmd = cmd % (port, host)
+        with open(os.path.expanduser('~/.ssh/known_hosts'), 'a+') as hosts:
+            keys = [k.replace('\n', '') for k in hosts.readlines()]
+            new_keys = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE
+                                        ).communicate()[0].split('\n')
+            for key in new_keys:
+                if [k for k in keys if k == key]:
```

Comment:
@ruiztulio 
This `if` avoid that each called of the function `_ssh_keyscan` adding the same keys to the `.ssh/known_hosts` file